src/Workleap.Extensions.OpenApi.Analyzers/CompareTypedResultWithAnnotationAnalyzer.cs

```diff
@@ -0,0 +1,279 @@
+                        ValidateAnnotationForTypeMismatch(context, methodSignatureStatusCodeToTypeMap, statusCodeValue, type, attribute);
+                    }
+                }
+            }
+        }
+
+        private Dictionary<int, ITypeSymbol> GetMethodReturnStatusCodeToTypeMap(ITypeSymbol methodReturnSignature)
+        {
+            var methodReturnSignatures = this.GetReturnStatusAndTypeFromMethodReturn(methodReturnSignature);
+            var methodSignatureStatusCodeToTypeMap = new Dictionary<int, ITypeSymbol>();
+            foreach (var returnValues in methodReturnSignatures)
+            {
+                if (!methodSignatureStatusCodeToTypeMap.ContainsKey(returnValues.statusCode))
+                {
+                    methodSignatureStatusCodeToTypeMap.Add(returnValues.statusCode, returnValues.symbol);
+                }
+            }
+
+            return methodSignatureStatusCodeToTypeMap;
+        }
+
+
+        private IEnumerable<(int statusCode, ITypeSymbol symbol)> GetReturnStatusAndTypeFromMethodReturn(ITypeSymbol typeSymbol)
+        {
+            if (typeSymbol is not INamedTypeSymbol namedTypeSymbol)
+            {
+                return Enumerable.Empty<(int, ITypeSymbol)>();
+            }
+
+            // Task<>
+            if (SymbolEqualityComparer.Default.Equals(namedTypeSymbol.ConstructedFrom, TaskOfTSymbol))
+            {
+                return this.GetReturnStatusAndTypeFromMethodReturn(namedTypeSymbol.TypeArguments[0]);
+            }
+
+            if (SymbolEqualityComparer.Default.Equals(namedTypeSymbol.ConstructedFrom, this.ValueTaskOfTSymbol))
+            {
+                return this.GetReturnStatusAndTypeFromMethodReturn(namedTypeSymbol.TypeArguments[0]);
+            }
+
+            // Result<OK, NotFound>
+            if (this.ResultTaskOfTSymbol.Any(symbol => SymbolEqualityComparer.Default.Equals(namedTypeSymbol.ConstructedFrom, symbol)))
+            {
+                return namedTypeSymbol.TypeArguments.SelectMany(this.GetReturnStatusAndTypeFromMethodReturn);
+            }
+
+            if (this._resultsToStatusCodeMap.TryGetValue(namedTypeSymbol.ConstructedFrom, out var statusCode))
+            {
+                // If there is a type, then return the type, otherwise return IResult type
+                return [(statusCode, namedTypeSymbol.TypeArguments.Length == 0 ? namedTypeSymbol : namedTypeSymbol.TypeArguments[0])];
+            }
+
+            return Enumerable.Empty<(int, ITypeSymbol)>();
+        }
+
+        private static void ValidateAnnotationForTypeMismatch(SymbolAnalysisContext context,
+            Dictionary<int, ITypeSymbol> methodSignatureStatusCodeToTypeMap, int statusCodeValue, ITypeSymbol type, AttributeData attribute)
+        {
+            if (attribute.ApplicationSyntaxReference is null)
+            {
+                return;
+            }
+
+            var attributeLocation = attribute.ApplicationSyntaxReference.GetSyntax().GetLocation();
```

Comment:
I'm pretty sure you can forward a CancellationToken to GetSyntax (context.CancellationToken).